src/main/java/com/exactpro/th2/act/actions/ActAction.java

```diff
@@ -56,46 +54,30 @@ protected boolean storeParentEvent() {
 	}
 
 	public void run(T details) {
-
 		RhSessionID sessionID = getSessionID(details);
 
-		ActResult actResult = new ActResult();
+		ActResult actResult = createActResult();
 		K frameworkContext = null;
-		EventID parentEventId = null;
-		EventID actionEvent = null;
-		
+		EventID eventId = null;
+
 		try {
 			frameworkContext = framework.newExecution(sessionID);
-			parentEventId = getParentEventId(details);
-			if (storeParentEvent()) {
-				Map<String, String> requestParams = convertRequestParams(details);
-				actionEvent = framework.createParentEvent(parentEventId, getName(), requestParams);
-				frameworkContext.setParentEventId(actionEvent);
-			} else {
-				frameworkContext.setParentEventId(parentEventId);
-			}
+			eventId = this.processAndGetEventId(frameworkContext, details);
```

Comment:
framework.newExecution is throwable in that case we will have NPE in catch block